### ISSUE TITLE
JVM Crash when using allocation profiling

### ISSUE THREAD

**Issue by RoySunnySean007** [jvm-profiling-tools/async-profiler#449]
Dear Async profiler experts,

When I use the following command to do allocation profiling, JVM crash, could u pls kindly help to have a look? Thx!!!
./profiler.sh -d 5 -e mprotect -f a.svg <java pid> 

Async profiler version: 1.8.6
Java version (I have installed debug symbol): 
openjdk version "1.8.0_282-debug"
OpenJDK Runtime Environment (build 1.8.0_282-debug-b08)
OpenJDK 64-Bit Server VM (build 25.282-b08-debug, mixed mode)

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (/builddir/build/BUILD/java-1.8.0-openjdk-1.8.0.282.b08-1.amzn2.0.1.x86_64/openjdk/hotspot/src/share/vm/runtime/mutex.cpp:336), pid=28535, tid=0x00007f8c6aa7d700
#  assert(w == 0 || w == _LBIT) failed: invariant
#
# JRE version: OpenJDK Runtime Environment (8.0_282-b08) (build 1.8.0_282-debug-b08)
# Java VM: OpenJDK 64-Bit Server VM (25.282-b08-debug mixed mode linux-amd64 compressed oops)
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# If you would like to submit a bug report, please visit:
#   https://forums.aws.amazon.com/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

---------------  T H R E A D  ---------------

Current thread (0x00007f8ca0001800):  JavaThread "Attach Listener" daemon [_thread_in_native, id=28595, stack(0x00007f8c6a97d000,0x00007f8c6aa7e000)]

Stack: [0x00007f8c6a97d000,0x00007f8c6aa7e000],  sp=0x00007f8c6aa7ae20,  free space=1015k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [libjvm.so+0xe64d7f]  VMError::report(outputStream*)+0xf2b
V  [libjvm.so+0xe661ff]  VMError::report_and_die()+0x475
V  [libjvm.so+0x6a35a6]  report_vm_error(char const*, int, char const*, char const*)+0xb4
V  [libjvm.so+0xbc273e]  Monitor::ILocked()+0x4a
V  [libjvm.so+0xbc29e9]  Monitor::ILock(Thread*)+0x79
V  [libjvm.so+0xbc39c6]  Monitor::lock_without_safepoint_check(Thread*)+0x5a
V  [libjvm.so+0xbc3a35]  Monitor::lock_without_safepoint_check()+0x23
C  [libasyncProfiler.so+0x21b25]  VM::loadMethodIDs(_jvmtiEnv*, JNIEnv_*, _jclass*)+0xb5
C  [libasyncProfiler.so+0x2248b]  VM::init(JavaVM_*, bool)+0x46b
C  [libasyncProfiler.so+0x22632]  Agent_OnAttach+0x22
V  [libjvm.so+0xa71a0d]  JvmtiExport::load_agent_library(AttachOperation*, outputStream*)+0x351
V  [libjvm.so+0x425a05]  attach_listener_thread_entry(JavaThread*, Thread*)+0x21e
V  [libjvm.so+0xe00b66]  JavaThread::thread_main_inner()+0x136
V  [libjvm.so+0xe00a11]  JavaThread::run()+0x1b3
V  [libjvm.so+0xc18d1d]  java_start(Thread*)+0x1ca
C  [libpthread.so.0+0x740b]  start_thread+0xdb
```

**Comment by debashisdeb** [comment]
@RoySunnySean007 Possible duplicate of https://github.com/jvm-profiling-tools/async-profiler/issues/226
Providing the `hs_err_pid.log` should help. 
Is it only happening with allocation profiling? Are you able to take a cpu profile?

**Comment by apangin** [comment]
You are running a *debug JVM*, not a production JVM with *debuginfo*. There is a substantial difference.

*Debug* JVM is not intended for end users - it is rather for JVM engineers to troubleshoot issues in the JVM itself. It runs a way slower, it has tons of additional assertions, and, to be honest, it has never been tested together with async-profiler. I guess the crash is a result of the debug build having different layout of JVM structures that async-profiler relies upon.

I'll look why async-profiler does not work with debug JVM, but anyway, it's better to switch to a normal JVM version (it should not have `debug` in `java -version` output).

So, what you need is to take a regular JDK package and [install](https://github.com/jvm-profiling-tools/async-profiler#installing-debug-symbols) additional symbols (debuginfo).

Or, even better, take an OpenJDK distribution that is profiler-friendly out of the box. My personal preference is [Liberica JDK](https://bell-sw.com/pages/downloads/) or [Zulu JDK](https://www.azul.com/downloads/?package=jdk).

**Comment by RoySunnySean007** [comment]
Thx @apangin ! I tried normal JVM version and it worked!

**Comment by apangin** [comment]
BTW, I've fixed async-profiler to work with debug builds of JVM.

**Comment by RoySunnySean007** [comment]
Thx @apangin !
